packages/ember/tests/component_registration_test.js

```diff
@@ -1,200 +1,172 @@
+    this.visit('/').then(() => {
+      let text = this.$('#wrapper').text().trim();
+      assert.equal(text, 'there goes watch him as he GOES', 'The component is composed correctly');
+    });
+  }
 
-  boot(() => {
-    appInstance.register('component:my-hero', Component.extend({
-      classNames: 'testing123',
-      layout: compile('watch him as he GOES')
-    }));
-  });
+  ['@test Late-registered components can be rendered with template registered on the container'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>hello world {{sally-rutherford}}-{{#sally-rutherford}}!!!{{/sally-rutherford}}</div>');
+    this.addComponent('sally-rutherford', {
+      ComponentClass: Component,
+      template: 'funkytowny{{yield}}'
+    });
 
-  equal(jQuery('#wrapper').text(), 'there goes watch him as he GOES', 'The component is composed correctly');
-});
+    this.visit('/').then(() => {
+      let text = this.$('#wrapper').text().trim();
+      assert.equal(text, 'hello world funkytowny-funkytowny!!!', 'The component is composed correctly');
+    });
+  }
 
-QUnit.test('Late-registered components can be rendered with template registered on the container', function() {
-  setTemplate('application', compile('<div id=\'wrapper\'>hello world {{sally-rutherford}}-{{#sally-rutherford}}!!!{{/sally-rutherford}}</div>'));
+  ['@test Late-registered components can be rendered with ONLY the template registered on the container'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>hello world {{borf-snorlax}}-{{#borf-snorlax}}!!!{{/borf-snorlax}}</div>');
 
-  boot(() => {
-    appInstance.register('template:components/sally-rutherford', compile('funkytowny{{yield}}'));
-    appInstance.register('component:sally-rutherford', Component);
-  });
+    this.addComponent('borf-snorlax', {
+      template: 'goodfreakingTIMES{{yield}}'
+    });
 
-  equal(jQuery('#wrapper').text(), 'hello world funkytowny-funkytowny!!!', 'The component is composed correctly');
-});
+    this.visit('/').then(() => {
+      let text = this.$('#wrapper').text().trim();
+      assert.equal(text, 'hello world goodfreakingTIMES-goodfreakingTIMES!!!', 'The component is composed correctly');
+    });
+  }
 
-QUnit.test('Late-registered components can be rendered with ONLY the template registered on the container', function() {
-  setTemplate('application', compile('<div id=\'wrapper\'>hello world {{borf-snorlax}}-{{#borf-snorlax}}!!!{{/borf-snorlax}}</div>'));
+  ['@test Assigning layoutName to a component should setup the template as a layout'](assert) {
+    assert.expect(1);
 
-  boot(() => {
-    appInstance.register('template:components/borf-snorlax', compile('goodfreakingTIMES{{yield}}'));
-  });
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>');
+    this.addTemplate('foo-bar-baz', '{{text}}-{{yield}}');
+    this.add('controller:application', Controller.extend({
```

Comment:
per the original test, this should be registered onto an autobooted application instance. the component `my-component` should be as well.